### PR TITLE
test: use non-deprecated gomock package & fix unit tests

### PR DESCRIPTION
**Pull request by jtieri**
the [github.com/golang/mock](https://github.com/golang/mock) package has been archived with the suggestion to use the forked version [here](https://github.com/uber-go/mock). this pr replaces the now abandoned repo in favor of the maintained fork and regenerates the mocks.

d72e83c94350433a1f194dbfaff97980b3a274a7 also introduced some breakage in the unit tests which i've fixed.

### CODE REVIEW

File: middleware/packet-forward-middleware/Makefile [hunk 1]
```diff
@@ -147,6 +147,7 @@ mocks: $(MOCKS_DIR)
 	mockgen -package=mock -destination=./test/mock/bank_keeper.go $(GOMOD)/router/types BankKeeper
 	mockgen -package=mock -destination=./test/mock/ics4_wrapper.go github.com/cosmos/ibc-go/v7/modules/core/05-port/types ICS4Wrapper
 	mockgen -package=mock -destination=./test/mock/ibc_module.go github.com/cosmos/ibc-go/v7/modules/core/05-port/types IBCModule
+	mockgen -package=mock -destination=./test/mock/channel_keeper.go $(GOMOD)/router/types ChannelKeeper
```
Comment (Reecepbcups) [inline middleware/packet-forward-middleware/Makefile:150]: Maybe add  `go install go.uber.org/mock/mockgen@v0.2.0` into the `mocks` target, or a comment to use this for mockgen now.

**Review by Reecepbcups — APPROVED** [review]

**Comment by Reecepbcups** [comment]
cc @jtieri this is actually approved now with me being added to the IBCApps team

File: middleware/packet-forward-middleware/Makefile (hunk 1 above)
Comment (jtieri) [reply middleware/packet-forward-middleware/Makefile:150]: good idea, i'll add this now!

**Comment by jtieri** [comment]
> cc @jtieri this is actually approved now with me being added to the IBCApps team

thanks! just getting back around to this, i'll add that line to the Makefile target for generating the mocks then get this merged.

**Pull request merged**